### ISSUE TITLE
Rule proposal: no-useless-fallback

### ISSUE THREAD

**Issue by BridgeAR** [sindresorhus/eslint-plugin-unicorn#3592]
### Description

I frequently see code that iterates over an optional object by creating an empty object in case the variable is not defined. That creates overhead and the loop should be placed inside an if statement instead.

The same applies to other spots, while I see it most frequent in for loops.

### Examples

```js
// ❌
for (const [key, value] of Object.entries(options.config ?? {})) {}

// ✅
if (options.config) {
  for (const [key, value] of Object.entries(options.config)) {}
}
```

### Proposed rule name

no-useless-fallback

**Comment by silverwind** [comment]
I actually prefer the `?? {}` and `?? []` forms and would actually propose the rule to allow the user to choose which form they want. Rule name could be `iteration-fallback-style`.

**Comment by BridgeAR** [comment]
We'd allocate an object and call the method on it, so we have an overhead for the shorter form

**Comment by silverwind** [comment]
That's ok, I think both forms are warranted.

BTW, If we're talking about such minor performance details, `for-of` is known to not be the fastest loop and you should probably use `for` 😆.
